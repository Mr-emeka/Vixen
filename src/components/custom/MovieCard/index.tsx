import React from 'react'


type IProps = {
    name?: string,
    url?: string,
    genres?: string[],
    premiered?: string,
}
const MovieCard: React.FC<IProps> = ({ name, url }) => {
    return <div className="card">
        <img className="card__img" alt={`card__${name}`} src={url} />
        <div className='card__content d-flex'>
            <h3 className="card__text">{name}</h3>
        </div>
    </div>
}

export default MovieCard;