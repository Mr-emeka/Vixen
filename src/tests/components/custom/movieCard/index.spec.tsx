import React from "react";
import { shallow } from 'enzyme';
import MovieCard from '../../../../components/custom/MovieCard';

describe('Movie Card', () => {
    it('should render correctly', () => {
        const Card = shallow(<MovieCard url="" name="movie 1" />);
        const img = Card.find('.card__img')
        const text = Card.find('.card__text')
        expect(text.text()).toEqual('movie 1')
        expect(img.props().alt).toEqual(`card__movie 1`)
        expect(Card).toMatchSnapshot();
    });
});