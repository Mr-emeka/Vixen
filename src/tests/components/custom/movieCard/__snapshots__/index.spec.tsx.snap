// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Renders Card 1`] = `ShallowWrapper {}`;

exports[`Movie Card should render correctly 1`] = `ShallowWrapper {}`;
